test/astNodeHelpers.ts

```diff
@@ -0,0 +1,118 @@
+import {ASTBool, ASTInt, ASTStringLit} from '../src/server/ast/literals'
+import {ASTBlock, ASTClass, ASTParameter, ASTParams, ASTReturn, ASTTemplate} from '../src/server/ast/statements'
+import {MangroveSymbol, SymbolType, SymbolTypes} from '../src/server/ast/symbolTable'
+import {ASTIntType, ASTNode} from '../src/server/ast/types'
+import {
+	ASTCallArguments,
+	ASTDottedIdent,
+	ASTIdent,
+	ASTIdentDef,
+	ASTTemplateArguments,
+	ASTTypeDecl,
+} from '../src/server/ast/values'
+import {Parser} from '../src/server/parser/parser'
+import {Token, TokenType} from '../src/server/parser/types'
+
+export function astIdent(ident: string, symbol?: SymbolTypes, type?: TokenType)
+{
+	return new ASTIdent(
+		Token.from(type ?? TokenType.ident, ident),
+		symbol === undefined ? undefined : new MangroveSymbol(ident, new SymbolType(symbol)),
```

Comment:
Could the check here just be `!symbol ?`, or how important is it to check for `undefined` explicitly?

Reply:
Since `SymbolTypes.invalid` is `0`, it's also `falsy`. Therefore explicit `undefined` is important

Reply:
Ahhh, ok. That noodle. Good reason :)